cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -153,14 +152,22 @@ func getAPIServerCommand(cfg *kubeadmapi.InitConfiguration) []string {
 	if cfg.Etcd.External != nil {
 		defaultArguments["etcd-servers"] = strings.Join(cfg.Etcd.External.Endpoints, ",")
 
-		// Use any user supplied etcd certificates
+		// Set the etcd certificates to the default locations.
+		defaultArguments["etcd-cafile"] = filepath.Join(cfg.CertificatesDir, kubeadmconstants.EtcdCACertName)
+		defaultArguments["etcd-certfile"] = filepath.Join(cfg.CertificatesDir, kubeadmconstants.APIServerEtcdClientCertName)
+		defaultArguments["etcd-keyfile"] = filepath.Join(cfg.CertificatesDir, kubeadmconstants.APIServerEtcdClientKeyName)
+
+		// If they are defined in the configuration, use that value instead.
 		if cfg.Etcd.External.CAFile != "" {
 			defaultArguments["etcd-cafile"] = cfg.Etcd.External.CAFile
 		}
-		if cfg.Etcd.External.CertFile != "" && cfg.Etcd.External.KeyFile != "" {
+		if cfg.Etcd.External.CertFile != "" {
```

Comment:
IMO external cert and key should appear in pair. I'd be good to assume they are configured in the same time. WDYT?

Reply:
The logic behind this change was that if a user sets one but not the other they likely want that one overridden and the default for the other one. 

Before, if a user set one and not the other, neither would have a value (and no warning/messaging) which seems a bit confusing. Perhaps we need a warning if only one is set? Something like "Only Certfile or Keyfile was set. Both must be set or both must be empty. Assuming defaults." ?

Reply:
>The logic behind this change was that if a user sets one but not the other they likely want that one overridden and the default for the other one.

SGTM with leaving your change as-is :)